packages/migrate/src/utils/getDatabaseVersionSafe.ts

```diff
@@ -0,0 +1,28 @@
+import Debug from '@prisma/debug'
+
+import { MigrateEngine } from '../MigrateEngine'
+
+const debug = Debug('prisma:cli')
+
+/**
+ * Retrieve the database version from the given schema or url.
+ * This function never throws, and was introduced to prevent circular dependencies in `@prisma/internals`.
+ */
+export async function getDatabaseVersionSafe(schemaOrUrl: string): Promise<string | undefined> {
```

Comment:
When reading the code, this name threw me off, but I guess it makes sense to describe what it is doing.